tests/test_kubernetes_tools.py

```diff
@@ -1422,6 +1422,41 @@ def test_get_boto_volume(
         else:
             assert volumes is None
 
+    @pytest.mark.parametrize(
+        "config_dict",
+        [
+            {"database_credentials": {"mysql": ["credential1", "credential2"]}},
+            {"database_credentials": {"mysql": ["credential3"]}},
+            {"database_credentials": {"mysql": []}},
+            {
+                "database_credentials": {
+                    "mysql": ["credential1"],
+                    "cassandra": ["credential4", "credential5"],
+                }
+            },
+            {},
+        ],
+    )
+    def test_get_database_credentials_secrets_volumes(self, config_dict):
+        deployment = KubernetesDeploymentConfig(
+            service="my-service",
+            instance="my-instance",
+            cluster="mega-cluster",
+            config_dict=config_dict,
+            branch_dict=None,
+            soa_dir="/nail/blah",
+        )
+
+        volumes = deployment.get_database_credentials_secrets_volumes()
+        if config_dict:
+            assert volumes is not None
+            total_volumes = 0
+            for dstore in config_dict["database_credentials"]:
+                total_volumes += len(config_dict["database_credentials"][dstore])
+            assert len(volumes) == total_volumes
```

Comment:
can we add the expected amount of volumes to the fixture? that would allow us to remove this for-loop and simplify this check to 
```
assert len(volumes) == expected_num_volumes
```

ideally we'd also split this up into two tests so that we can get rid of the conditional in this test, but it looks like I missed asking for that refactor for boto_keys so it's not the end of the world if the conditional logic stays since it's not inconsistent with other tests (and worst case, an enterprising yelper can simply refactor all these cases in the future)

Reply:
++ will simplify and split

enterprising refactorers are gems. if we find one, I'll name my firstborn after them